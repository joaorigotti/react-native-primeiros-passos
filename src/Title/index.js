import React from 'react';
import {
  StyleSheet,
  Text,
} from 'react-native';
import { getWidth } from '../Utils';

const style = ({size, bold, color}) => {
  return StyleSheet.create({
    title: {
      fontFamily: 'RobotoSlab-Regular',
      fontSize: size,
      fontWeight: bold ? 'bold' : 'normal',
      color: color || '#4d4d4d',
    },
  });
};

const Photo = props => {
  const _props = {
    size: props.size,
    bold: props.bold,
    color: props.color,
  };
  const styles = style(_props);

  return <Text style={styles.title}>{props.children}</Text>;
}

export default Photo
