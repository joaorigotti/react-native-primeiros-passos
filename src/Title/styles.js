import { StyleSheet } from 'react-native';
import * as cs from '../CommonStyles';

export default ({size, bold, color}) => {
  return StyleSheet.create({
    title: {
      fontFamily: cs.MAIN_FONT,
      fontSize: size,
      fontWeight: bold ? 'bold' : 'normal',
      color: color || cs.COLOR.gray30,
    },
  });
};