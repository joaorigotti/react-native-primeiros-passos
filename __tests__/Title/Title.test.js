import React from 'react';
import { shallow } from 'enzyme';
import Title from '../../src/Title';

describe('<Title />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <Title />
    );
    
    expect(wrapper).toMatchSnapshot();
  });
});