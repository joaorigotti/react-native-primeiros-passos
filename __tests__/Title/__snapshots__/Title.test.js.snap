// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Title /> renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
  style={
    Object {
      "color": "#4d4d4d",
      "fontFamily": "RobotoSlab-Regular",
      "fontSize": undefined,
      "fontWeight": "normal",
    }
  }
/>
`;
